air/tests/test_module/instructions/match_.rs

```diff
@@ -234,3 +234,39 @@ fn match_with_two_xors() {
 
     assert_eq!(actual_trace.pop().unwrap(), expected_executed_call_result);
 }
+
+#[test]
+fn test_match_with_booleans() {
+    let result_setter_peer_id = "result_setter_peer_id";
+    let mut result_setter = create_avm(
+        set_variable_call_service(serde_json::json!({"success": true})),
+        result_setter_peer_id,
+    );
+
+    let echo_peer_id = "echo_peer_id";
+    let mut echo_peer = create_avm(echo_call_service(), echo_peer_id);
+
+    let script = format!(
+        r#"
+        (seq
+            (call "{0}" ("" "") ["set_result"] result)
+            (seq
+                (xor
+                    (match result.$.success! true
```

Comment:
Does `!` matter here?

Reply:
If not, maybe modify the test so it tests both `$.success!` and `$.success`?

Reply:
I'd be sleeping better if I know that this behavior is captured in tests :)

Reply:
No, `!` is ignored by lambda, it's possible to write it for backward compatibility, but it doesn't really make sense.

Reply:
Ah, so it's ignored everywhere? That's great :) No need for tests modification then, agree.

Reply:
Yep, it ignored everywhere